components/datetime/src/provider.rs

```diff
@@ -0,0 +1,232 @@
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Month(fields::Month::Format),
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Day(fields::Day::DayOfMonth),
+                length: FieldLength::Short,
+            },
+        ])),
+        Pattern(Cow::Borrowed(&[
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Day(fields::Day::DayOfMonth),
+                length: FieldLength::TwoDigit,
+            }),
+            PatternItem::Literal(Cow::Borrowed("/")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Month(fields::Month::Format),
+                length: FieldLength::TwoDigit,
+            }),
+            PatternItem::Literal(Cow::Borrowed("/")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Year(fields::Year::Calendar),
+                length: FieldLength::Short,
+            }),
+        ])),
+    ),
+    (
+        Skeleton(Cow::Borrowed(&[
+            Field {
+                symbol: FieldSymbol::Year(fields::Year::Calendar),
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Month(fields::Month::Format),
+                length: FieldLength::Abbreviated,
+            },
+            Field {
+                symbol: FieldSymbol::Day(fields::Day::DayOfMonth),
+                length: FieldLength::Short,
+            },
+        ])),
+        Pattern(Cow::Borrowed(&[
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Day(fields::Day::DayOfMonth),
+                length: FieldLength::Short,
+            }),
+            PatternItem::Literal(Cow::Borrowed(" ")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Month(fields::Month::Format),
+                length: FieldLength::Abbreviated,
+            }),
+            PatternItem::Literal(Cow::Borrowed(" ")),
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Year(fields::Year::Calendar),
+                length: FieldLength::Short,
+            }),
+        ])),
+    ),
+    (
+        Skeleton(Cow::Borrowed(&[
+            Field {
+                symbol: FieldSymbol::Hour(fields::Hour::H23),
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Minute,
+                length: FieldLength::Short,
+            },
+            Field {
+                symbol: FieldSymbol::Second(fields::Second::Second),
+                length: FieldLength::Short,
+            },
+        ])),
+        Pattern(Cow::Borrowed(&[
+            PatternItem::Field(Field {
+                symbol: FieldSymbol::Hour(fields::Hour::H23),
+                length: FieldLength::TwoDigit,
+            }),
+            PatternItem::Literal(Cow::Borrowed(":")),
```

Comment:
`":"` should also come from the data provider (same a few lines below)

Reply:
It will. The code in `provider.rs` is the minimal mock of what we'll get from `DataProvider`.